src/Data/List/Compat.hs

```diff
@@ -70,4 +129,15 @@ uncons                  :: [a] -> Maybe (a, [a])
 uncons []               = Nothing
 uncons (x:xs)           = Just (x, xs)
 
+-- | A strictly accumulating version of 'scanl'
+{-# NOINLINE [1] scanl' #-}
+scanl'           :: (b -> a -> b) -> b -> [a] -> [b]
```

Comment:
Wow, I didn't even notice that `scanl'` was added in `base-4.8.0.0`. Nice catch!